source/models/finding-records.md

```diff
@@ -1,16 +1,201 @@
 The Ember Data store provides a simple interface for finding records of a single
-type through the `store` object's `find` method. Internally, the `store`
-uses `find`, `findAll`, and `findQuery` based on the supplied arguments.
+type through the `store` object's `find` method.
 
 The first argument to `store.find()` is always the record type. The optional second
 argument determines if a request is made for all records, a single record, or a query.
 
 ### Finding All Records of a Type
 
-```javascript
-var posts = this.store.find('post'); // => GET /posts
+```app/routes/posts.js
+export default Ember.Route.extend({
+  model: function() {
+    return this.store.find('post').then(function(posts) {
+      // Optionally do something with posts
+      return posts;
+    });
+  }
+});
 ```
 
+`find` returns a `DS.PromiseArray` that fulfills to a `DS.RecordArray`
+containing all the records returned by the adapter.
+
+#### Server Response to a FindAll request
+
+When using the default `RESTAdapter` and `RESTSerializer` Ember Data
+will perform the following steps to attempt to request the `post` data
+from your server.
+
+First the `RESTAdapter` will attempt to build the URL it will use to
+request the `post` data. It does this by pluralizing and camelCasing
+the record type name. For the above example, `post` would become the
+url `/posts`. The `RESTAdapter` will then make a `GET` XHR (or ajax)
+request to the `/posts` url.
+
+Once the `GET /posts` requests is fulfilled the adapter will return the
+data from the response to Ember Data's store. The store will use the
+`RESTSerializer` to extract the data and normalize it into a form the
+store can understand.
+
+Given the following model definitions:
+
+```app/models/post.js
+export default DS.Model.extend({
+  title: DS.attr('string'),
+  tag: DS.attr('string'),
+  comments: DS.hasMany('comment')
+});
+```
+
+```app/models/comment.js
+export default DS.Model.extend({
+  body: DS.attr('string'),
+  post: DS.belongsTo('post')
+});
+```
+
+The `RESTSerializer` would expect the response from your server to
+look like the following json:
+
+```json
+{
+  "post": [{
+    "id": 1,
+    "title": "Rails is omakase",
+    "tag": "rails",
+    "comments": [1, 2]
+  }, {
+    "id": 2,
+    "title": "I'm Running to Reform the W3C's Tag",
+    "tag": "w3c",
+    "comments": [3]
+  }],
+  "comments": [{
+    "id": 1,
+    "body": "FIRST",
+    "post": 1,
+  }, {
+    "id": 2,
+    "body": "Rails is unagi",
+    "post": 1
+  }, {
+    "id": 3,
+    "body": "Good luck!",
+    "post": 3
+  }]
+}
+```
+
+One thing you may notice is this response contains the data for three
+records instead of only the `post` record that was originally
+requested. This is a pattern called "sideloading". Sideloading allows
```

Comment:
This sentence doesn't really make sense, we're doing `find('post')` so we're not requesting a single `post` record.